pkg/apiserver/profiling/service.go

```diff
@@ -131,8 +168,79 @@ func (s *Service) start(c *gin.Context) {
 		taskGroup.State = TaskStateFinish
 		s.db.Save(taskGroup.TaskGroupModel)
 	}()
+	return taskGroup, nil
+}
 
-	c.JSON(http.StatusOK, taskGroup.TaskGroupModel)
+func (s *Service) autoCollect(req *StartRequest) {
+	defer s.wg.Done()
+	interval := time.Duration(req.CollectionInterval+req.DurationSecs) * time.Second
+	timer := time.NewTimer(interval)
+	defer timer.Stop()
+
+	for {
+		if atomic.LoadUint32(&s.isAutoCollectRunning) == 0 {
+			return
+		}
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-timer.C:
+			_, _ = s.collect(req)
+			timer.Reset(interval)
+		default:
+		}
+	}
+}
+
+// @Summary Start auto profiling
+// @Description Start auto profiling
+// @Produce json
+// @Param req body StartRequest true "auto profiling request"
+// @Security JwtAuth
+// @Success 200 {string} string "success"
+// @Failure 400 {object} utils.APIError
+// @Router /profiling/auto/start [post]
+func (s *Service) autoStart(c *gin.Context) {
+	if atomic.LoadUint32(&s.isAutoCollectRunning) > 0 {
```

Comment:
Would it be better to use `CompareAndSwap`?

Reply:
Yes, we need to use CAS.